storm-server/src/main/java/org/apache/storm/metric/StormMetricsRegistry.java

```diff
@@ -73,6 +130,12 @@ private static void startMetricsReporter(PreparableReporter reporter, Map<String
     }
 
     private static <T extends Metric> T register(String name, T metric) {
+        if (source != null && !name.startsWith(source)) {
```

Comment:
I think it would be better to register with a Daemon type enum and metric name.  This would allows metrics that are daemon specific and general to work for a given daemon.

Reply:
I was originally thinking so but then I found out there are some other components not specified as 'daemon' in DaemonType but also using metrics, such as logviewer and RocksDB. I wasn't sure how to handle the case so internally I convert them all to String. It's possible to register both with a Daemon and with a name in String.

Reply:
Thanks for the clarification. Anyone else also has suggestions in the implementation?